### PR TITLE
History of the problem changes

### LINKED ISSUE #395: History of the problem changes
Display as description change on a goal in issues. Also need to create a model for changes.

### PR DESCRIPTION
**Pull request by Katrin-kudryash**
## PR includes

-   [x] Feature

## Related issues

Resolve #395

Related issues #394 , #393

### CODE REVIEW

**Comment by Katrin-kudryash** [comment]
Untitled 14

**Comment by Katrin-kudryash** [comment]
Untitled1

**Comment by Katrin-kudryash** [comment]
https://github.com/taskany-inc/hire/assets/81121562/30782b2a-a0d7-4d39-914e-4eb076db4f7e

**Comment by Katrin-kudryash** [comment]
Untitled 13

File: src/modules/problemTypes.ts [hunk 1]
```diff
@@ -64,3 +69,5 @@ export type ProblemWithRelationsAndProblemSection = ProblemFindManyWithAuthorAnd
     CheckIsAvailableProblemType;
 
 export type ProblemWithRelations = ExtractPrismaTypeFromArray<typeof problemMethods.getList>;
+
+export type ProblemHistoryWithUser = ProblemHistory & { user: User };
```
Comment (nyet-ty) [inline src/modules/problemTypes.ts:73]: I'm sure that this should be higher than ProblemFindManyWithAuthorAndTagsAndFavourite

**Review by nyet-ty — APPROVED** [review]

**Comment by Katrin-kudryash** [comment]
Untitled 23
b5e8b31">

**Comment by Katrin-kudryash** [comment]
Untitled 23

**Comment by Katrin-kudryash** [comment]
Untitled 24

**Comment by Katrin-kudryash** [comment]
Untitled

File: src/components/ProblemHistoryCard/ProblemHistoryCard.tsx [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+                <VisibleContainer>
+                    <Button
+                        ghost={!viewProblemHistoryDescription}
+                        outline={viewProblemHistoryDescription}
+                        iconRight={<IconDividerLineOutline size="xs" />}
+                        onClick={handlerViewProblemHistoryDescription}
+                    />
+                </VisibleContainer>
+            </CardTitleContainer>
+            {viewProblemHistoryDescription && (
```
Comment (nyet-ty) [inline src/components/ProblemHistoryCard/ProblemHistoryCard.tsx:94]: nullable

**Review by nyet-ty — APPROVED** [review]

**Comment by Katrin-kudryash** [comment]
Untitled 3

**Comment by Katrin-kudryash** [comment]
Untitled 4

**Comment by Katrin-kudryash** [comment]
Untitled 9

File: src/modules/problemMethods.ts [hunk 3]
```diff
@@ -266,14 +272,42 @@ const getList = async (
...
+                subject: 'description',
+                user: { connect: { id: authorId } },
+                problem: { connect: { id: data.problemId } },
+                previousValue: problem.description,
+                nextValue: data.description,
+            },
+        });
+    }
+
+    if (data.solution !== problem.solution) {
```
Comment (9teen90nine) [inline src/modules/problemMethods.ts:299]: should be a single transaction with problem.update (since it's useless if update fails)

**Review by 9teen90nine — APPROVED** [review]

**Pull request merged**